Feature/skills

hilvitzs opened the pull request.

semantic-release-bot committed:
chore(release): 1.0.0 [skip ci]

# 1.0.0 (2019-03-07)

### Features

* **commitizen:** added commitizen support locally ([46658fc](https://github.com/buildit/dev-skill-entry/commit/46658fc))
* **semantic-release:** support for semantic release on `master` ([4919638](https://github.com/buildit/dev-skill-entry/commit/4919638))

Commit:
feat(database service): add database service

send user info to database, get user info from database, send user skills to database, get user
skills from database

Commit:
test(login.spec.ts, register.spec.ts): add tests

add login.spec.ts tests and register.spec.ts tests

Commit:
test(add tests): add tests for databaseService, skillsPage, skillsCard

Commit:
test(skill-page.spec.ts): add tests for skill page component

Commit:
test(nav.component.spec.ts): add register test

Commit:
refactor(skills-page): add json logic to setskills and get skills

Commit:
fix(skills page spec): fix skills page test

ashblue commented:
@hilvitzs Looks like console logs are causing travis to fail.

ashblue commented on src/app/app.module.ts:
```diff
@@ -42,7 +43,7 @@ import { RegisterComponent } from './materials/organisms/register/register.compo
     HttpClientModule,
   ],
-  providers: [],
+  providers: [AngularFirestore],
```
You should probably substitute the AngularFirestore with an empty object that has the methods you need. Live methods from a 3rd party API can be pretty error prone to version upgrades and dependency issues.

ashblue commented on src/app/materials/molecules/skill-card/skill-card.component.spec.ts:
```diff
@@ -23,10 +23,26 @@ describe('SkillCardComponent', () => {
   });
+
+  it('should have a title value', () => {
+    expect(fixture.nativeElement.querySelector('mat-card-title').innerText).toEqual('ReactJS');
```
Good use of fast and simple testing to execute TDD.

ashblue commented on src/app/materials/organisms/register/register.component.ts:
```diff
@@ -14,18 +15,33 @@ export class RegisterComponent {
+        uid: resp.user.uid,
+      };
+
+      this.data.setUser(userInfo);
```
Can't yo ujust pass in `resp.user` instead?

hilvitzs replied:
I tried that, unfortunately, it says that I can't submit that resp.user to firebase because it is a custom Q object. Object.assign doesn't change it so I figured if I'm cleaning it up anyway I might as well just grab the info I need.

ashblue commented on src/app/materials/pages/skills/skills-page.component.spec.ts:
```diff
@@ -1,15 +1,106 @@
+    };
+
+    const store = {};
+    const mockLocalStorage = {
```
If we end up using this again we should move it into a `shared` folder. Local storage mocks end up coming up again usually.

ashblue commented on src/app/materials/pages/skills/skills-page.component.spec.ts:
```diff
@@ -1,15 +1,106 @@
+    const store = {};
+    const mockLocalStorage = {
+      getItem: (key: string): string => {
+        return key in store ? store[key] : null;
```
Can we switch the ternary operator to an if? They're high risk for merge conflicts.

ashblue reviewed (approved):
This looks really good. Some minor comments in my review I'd recommend fixing before merging. 

Can you turn on https://palantir.github.io/tslint/rules/no-consecutive-blank-lines/ with the linter? Noticing some repeated blank lines. You should be able to run the auto fixer from the terminal to fix any errors (able to code pair with you on that).

Commit:
refactor(skills-page.spec.ts): change ternary to if statement

Commit:
refactor: remove extra empty lines and change tslint config rules

Commit:
refactor(setuser,getuser): remove setUser and getUser

realized that setUser and getUser are not necessary, user info is automatically stored

Commit:
refactor(databaseservice): remove database service methods

Commit:
refactor(skillspagespec): change doc return for test

Commit:
refactor(skills-page-component): remove console.log

Commit:
refactor(skills-page-component): remove console.log

Commit:
Merge branch 'master' of github.com:buildit/dev-skill-entry into feature/skills
